addons/website_event_questions/models/event_event.py

```diff
@@ -32,37 +33,28 @@ def _compute_question_ids(self):
 
         When synchronizing questions:
 
-          * lines that no answer are removed;
+          * lines with no registered answers are removed;
           * type lines are added;
         """
         if self._origin.question_ids:
             # lines to keep: those with already given answers
-            questions_tokeep_ids = self.env['event.registration.answer'].search(
-                [('question_id', 'in', self._origin.question_ids.ids)]
-            ).question_id.ids
+            questions_tokeep = self.env['event.registration.answer'].search([
+                ('question_id', 'in', self._origin.question_ids.ids),
+            ]).question_id
         else:
-            questions_tokeep_ids = []
+            questions_tokeep = None
+
         for event in self:
             if not event.event_type_id and not event.question_ids:
                 event.question_ids = False
                 continue
 
-            if questions_tokeep_ids:
-                questions_toremove = event._origin.question_ids.filtered(lambda question: question.id not in questions_tokeep_ids)
-                command = [(3, question.id) for question in questions_toremove]
+            if questions_tokeep:
+                questions_toremove = event._origin.question_ids - questions_tokeep
+                event.question_ids = [Command.unlink(q.id) for q in questions_toremove]
             else:
-                command = [(5, 0)]
-            if event.event_type_id.question_ids:
-                command += [
-                    (0, 0, {
-                        'title': question.title,
-                        'question_type': question.question_type,
-                        'sequence': question.sequence,
-                        'once_per_order': question.once_per_order,
-                        'answer_ids': [(0, 0, {
-                            'name': answer.name,
-                            'sequence': answer.sequence
-                        }) for answer in question.answer_ids],
-                    }) for question in event.event_type_id.question_ids
-                ]
-            event.question_ids = command
+                event.question_ids = [Command.clear()]
+
+            # copy questions so changes in the event don't affect the event type
+            for question in event.event_type_id.question_ids:
+                event.question_ids += question.copy({'event_type_id': False})
```

Comment:
Meh ... don't like this hardcoded copy, it is inefficient and not the way we want to use the ORM. Would say that either we find a way to propagate translations, either we consider it as a framework limitation. But I don't think we are going to create sequentially in a loop just for that.

Reply:
Also, I doubt anyone's creating events that require 20,000 questions on registration, so I don't think this is the place to be doing premature optimization.

Reply:
As discussed, code of 'copy_translation' is a bit too low-level (and anyway sequential) to be called manually anyway... so we can't easily copy translations in batch, hence ok for the 'copy' to keep code simple.